ehr/sync/task.go

```diff
@@ -34,3 +37,32 @@ func GetClinicId(data map[string]interface{}) (string, error) {
 	}
 	return clinicId, nil
 }
+
+func ScheduleNextExecution(tsk *task.Task) {
+	if tsk.HasError() {
+		tsk.RepeatAvailableAfter(OnErrorAvailableAfterDuration)
+		return
+	}
+
+	cadence := GetCadence(tsk.Data)
+	if cadence == nil {
+		tsk.RepeatAvailableAfter(DefaultCadence)
+	} else if *cadence != 0 {
+		tsk.RepeatAvailableAfter(*cadence)
+	} else {
+		tsk.SetCompleted()
+	}
+}
+
+func GetCadence(data map[string]interface{}) *time.Duration {
+	cadence := data["cadence"].(string)
```

Comment:
Uck. I get why this is here, but it's unfortunate. :( I guess someday we could update the task service now that Generics in Go are a thing.